pallet/pallet-codegen/src/main/java/com.strategyobject.substrateclient.pallet/StorageProcessor.java

```diff
@@ -0,0 +1,440 @@
+import com.strategyobject.substrateclient.storage.*;
+import com.strategyobject.substrateclient.types.FixedBytes;
+import com.strategyobject.substrateclient.types.Size;
+import lombok.NonNull;
+import lombok.val;
+import lombok.var;
+
+import javax.lang.model.element.*;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.TypeMirror;
+import java.util.List;
+import java.util.Objects;
+
+import static com.strategyobject.substrateclient.common.codegen.AnnotationUtils.suppressWarnings;
+import static com.strategyobject.substrateclient.pallet.Constants.*;
+
+class StorageProcessor extends PalletMethodProcessor {
+    private static final String VALUE_READER = "valueReader";
+    private static final String KEY_PROVIDER = "keyProvider";
+    private static final String STORAGE_MAP = "storage";
+    private static final String HASHERS = "hashers";
+    private static final String INTERNAL_SUFFIX = "Internal";
+
+    public StorageProcessor(@NonNull TypeElement palletElement) {
+        super(palletElement);
+    }
+
+    @Override
+    void process(@NonNull String palletName,
+                 @NonNull ExecutableElement method,
+                 TypeSpec.@NonNull Builder typeSpecBuilder,
+                 MethodSpec.Builder constructorBuilder,
+                 @NonNull ProcessorContext context) throws ProcessingException {
+        val annotation = AnnotationUtils.getAnnotationMirror(method, Storage.class);
+        if (annotation == null) {
+            return;
+        }
+
+        validate(method, annotation, context);
+
+        createBackField(typeSpecBuilder, method);
+        typeSpecBuilder.addMethod(privateMethod(palletName, method, annotation, context));
+        assignBackFieldInConstructor(constructorBuilder, method);
+        typeSpecBuilder.addMethod(publicMethod(method));
+    }
+
+    private void createBackField(TypeSpec.Builder typeSpecBuilder, ExecutableElement method) {
+        typeSpecBuilder.addField(
+                TypeName.get(method.getReturnType()),
+                method.getSimpleName().toString(),
+                Modifier.FINAL, Modifier.PRIVATE
+        );
+    }
+
+    private void assignBackFieldInConstructor(MethodSpec.Builder constructorBuilder, ExecutableElement method) {
+        constructorBuilder.addStatement("this.$1L = $1L$2L()",
+                method.getSimpleName().toString(),
+                INTERNAL_SUFFIX);
+    }
+
+    private MethodSpec publicMethod(ExecutableElement method) {
+        val returnType = method.getReturnType();
+
+        return MethodSpec.methodBuilder(method.getSimpleName().toString())
+                .addModifiers(Modifier.PUBLIC)
+                .addAnnotation(Override.class)
+                .returns(TypeName.get(returnType))
+                .addStatement("return this.$L", method.getSimpleName().toString())
+                .build();
+    }
+
+    private MethodSpec privateMethod(String palletName,
```

Comment:
this name is a bit misleading. I expected it to generate some internal method that it triggered by the public method defined later. But it turned out to be a some sort of `initializer`

Reply:
renamed to `backFieldInitializer`